RFCs/0020-User-Specified-TestAdapter-Lookup.md

```diff
@@ -0,0 +1,53 @@
+# 0020 User Specified Test Adapter Lookup
+
+## Summary
+This note outlines the proposed changes for lookup and initialization of the test adapters, based on User input.
+
+### Current Adapter Lookup
+
+In the current model, when ever a Run/Discovery call is made from IDE to TestPlatform, the test adapter initialization happens in multiple phases listed below
+
+* InitializeExtensions(): In this call, the IDE passes list of all adapters it finds in current session, these include adapters referenced via nuget in entire solution, & adpaters acquired via Vsix(e.g. GoogleTestAdapter, BoostAdapters, etc..)
+* TestAdapterPath: After InitializeExtensions call, TestPlatform then looks into adapters present in TestAdapterPath mentioned in RunSettings, & adds adapters to existing list.
+* TestPlatform then adds default Adapters from **Extensions** directory to current list.
+* Finally platform looks in adapters present in test source directory, & adds them to adapter list.
+
+Once adapters are gathered from all the possible locations, this list is passed to testhost process, which then loads all these adapters, & passes test source(s) to all the adapters.
+
+Following the above adapter look up logic, we pass down test sources to adapters which have no business running them, for e.g. passing a managed test source to a native adapter(GoogleTestAdapter, BoostAdapters, etc.), & vice-versa.
+
+## Motivation
+
+In .NetCore scenario, we only pick adapters from test Source location, this helps
+* Testplatform, as it knows where to look for adapters
+* Test Framework/Adapter writers, they know where to drop adapters so they are consumed by platform,& finally
```

Comment:
Motivation : Select the correct adapter for the run. Most managed adapters are available in the test source directory, hence can be picked up from later for the test run.

Proposed Changes : Introduce a setting to pick adapter from the source location.

These points should come out as pros, #1 is redundant.